core/synch.c

```diff
@@ -2267,4 +2267,5 @@ detach_on_permanent_stack(bool internal, bool do_cleanup, dr_stats_t *drstats)
     SELF_PROTECT_DATASEC(DATASEC_RARELY_PROT);
     dynamo_detaching_flag = LOCK_FREE_STATE;
     EXITING_DR();
+    options_detach();
```

Comment:
Better to call this before EXITING_DR which is supposed to be last.

Reply:
EXITING_DR ends up checking options which are reset during options_detach, so I think it needs to go after that point if we want to respect all of the flags until the very end.

options_detach simply overwrites some memory, there shouldn't be any complex interactions here afaict. Should I document inside options.c that options_detach must remain very simple since it basically occurs after detach has finished?

Reply:
I see, ok.  +1 to comment.